donald_wade/week_02/Thursday/main.js

```diff
@@ -0,0 +1,153 @@
+var $blob = $("img");
+
+setInterval(infiniteWalk, 50);
+var left = 0;
+$blob.css("left", left);
+function infiniteWalk() {
+  var $blob = $("img");
+  var widthOfGif = 296;
+  $blob.css("left", left)
+  left += 10;
+  var numPos = parseInt($blob.css("left"));
+  if (numPos >= window.innerWidth - widthOfGif) {
+    $blob.class("cat");
+    setInterval(goBackwards, 100);
+
+  }
+}
+
+
+
+
+function goBackwards() {
+  left -= 10;
+}
+
+
+
+
+
+
+// setInterval(catWalkBackwards, 50);
+
+// var left = 0;
+// function infiniteCatWalk() {
+//   $catGif.("left") = left + 'px';
+//   console.log($catGif);
+// }
+
+
+
+
+// setInterval(FWBW, 50);
+
+function FWBW() {
+  catGif.style.left = left + 'px';
+  console.log(catGif.style.left);
+  left += 10;
+  var numericalPosition = parseInt(catGif.style.left);
+  console.log(numericalPosition);
+  if (numericalPosition >= window.innerWidth - 296) {
+    console.log(fromRight);
+    catGif.className = "cat";
+    catGif.style.left = fromRight + 'px';
+    console.log(catGif.style.left);
+    fromRight -= 10;
+    catGif.style.left = fromRight + 'px';
+    console.log(fromRight);
+    var numericalPosition = parseInt(catGif.style.left);
+    console.log(numericalPosition);
+    console.log(catGif.style.left);
+    if (numericalPosition <= 10) {
+      console.log("Kaboom");
+      left = 0;
+      fromRight = window.innerWidth - 296;
+      catGif.className = "";
+
+    }
+  }
+}
+
+
+function basicCatWalk() {
+  catGif.style.left = left + 'px';
+  console.log(catGif.style.left);
+  left += 10;
+  var numericalPosition = parseInt(catGif.style.left);
+  console.log(numericalPosition);
+  if (numericalPosition >= window.innerWidth - 296) {
+    clearInterval(walkingForwards);
+  }
+}
+
+function catMoonWalk() {
+  catGif.style.left = left + 'px';
+  console.log(catGif.style.left);
+  left += 10;
+  var numericalPosition = parseInt(catGif.style.left);
+  console.log(numericalPosition);
+  if (numericalPosition >= window.innerWidth - 296) {
+    left -= 10;
+  }
+}
```

Comment:
basicCatWalk and catMoonWalk are almost identical functions, you might want to make it one reusable function and the only part that is changing can become a separate small function